ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java

```diff
@@ -215,6 +222,9 @@ public void close() {
       } finally {
         contentLock.unlock();
         restRequestMetricsTracker.recordMetrics();
+        if (digestCalculationTimeInMs >= 0) {
```

Comment:
Put it here because the request could error out before it reaches whoever calls `getDigest()`. You could argue that the metric might not be required in that case but the approach is consistent with the rest of our code where we record metrics in both the happy and unhappy paths.